cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/Capacity.java

```diff
@@ -0,0 +1,108 @@
+
+import io.strimzi.api.kafka.model.KafkaSpec;
+import io.strimzi.api.kafka.model.balancing.BrokerCapacity;
+import io.strimzi.api.kafka.model.storage.EphemeralStorage;
+import io.strimzi.api.kafka.model.storage.JbodStorage;
+import io.strimzi.api.kafka.model.storage.PersistentClaimStorage;
+import io.strimzi.api.kafka.model.storage.SingleVolumeStorage;
+import io.strimzi.api.kafka.model.storage.Storage;
+
+import java.util.List;
+
+import static io.strimzi.operator.cluster.model.StorageUtils.parseMemory;
+
+public class Capacity {
+    public static final long DEFAULT_BROKER_DISK_MIB_CAPACITY = 100_000;  // in MiB
+    public static final int DEFAULT_BROKER_CPU_UTILIZATION_CAPACITY = 100;  // as a percentage (0-100)
+    public static final int DEFAULT_BROKER_INBOUND_NETWORK_KIB_PER_SECOND_CAPACITY = 10_000;  // in KiB/s
+    public static final int DEFAULT_BROKER_OUTBOUND_NETWORK_KIB_PER_SECOND_CAPACITY = 10_000;  // in KiB/s
+
+    private Long diskMiB;
+    private Integer cpuUtilization;
+    private Integer inboundNetworkKiBPerSecond;
+    private Integer outboutNetworkKibPerSecond;
+
+    public Capacity(KafkaSpec spec) {
+        BrokerCapacity bc = spec.getCruiseControl().getBrokerCapacity();
+
+        this.diskMiB = bc != null && bc.getDiskMiB() != null ? bc.getDiskMiB() : generateDiskCapacity(spec.getKafka().getStorage());
+        this.cpuUtilization = bc != null && bc.getCpuUtilization() != null ? bc.getCpuUtilization() : DEFAULT_BROKER_CPU_UTILIZATION_CAPACITY;
+        this.inboundNetworkKiBPerSecond = bc != null && bc.getInboundNetworkKiBPerSecond() != null ? bc.getInboundNetworkKiBPerSecond() : DEFAULT_BROKER_INBOUND_NETWORK_KIB_PER_SECOND_CAPACITY;
+        this.outboutNetworkKibPerSecond = bc != null && bc.getOutboundNetworkKiBPerSecond() != null ? bc.getOutboundNetworkKiBPerSecond() : DEFAULT_BROKER_OUTBOUND_NETWORK_KIB_PER_SECOND_CAPACITY;
+    }
+
+    /**
+     * Generate diskMiB capacity configuration from the supplied storage configuration
+     *
+     * @param storage Storage configuration for Kafka cluster
+     * @return Disk capacity configuration as a Long
+     */
+    public static Long generateDiskCapacity(Storage storage) {
+        if (storage instanceof PersistentClaimStorage) {
+            return getSizeInMiB(((PersistentClaimStorage) storage).getSize());
+        } else if (storage instanceof EphemeralStorage) {
+            if (((EphemeralStorage) storage).getSizeLimit() != null) {
+                return getSizeInMiB(((EphemeralStorage) storage).getSizeLimit());
+            } else {
+                return DEFAULT_BROKER_DISK_MIB_CAPACITY;
+            }
+        } else if (storage instanceof JbodStorage) {
+            List<SingleVolumeStorage> volumeList = ((JbodStorage) storage).getVolumes();
+            long size = 0;
+            for (SingleVolumeStorage volume : volumeList) {
+                size += generateDiskCapacity(volume);
+            }
+            return size;
```

Comment:
Cruise Control does distinguish the disks as having individual sizes using the adminClient. 

However, for balancing partitions between disks of the same broker, Cruise Control needs us to explicitly state the capacity of each disk here, which I have not done yet. This will require some extra logic.

Reply:
So will that extra logic be in this PR, or a future one?

Reply:
A future one